2023/did-wg.html

```diff
@@ -337,13 +315,18 @@ <h2>Success Criteria</h2>
 more implementations interoperating with each other. In order to advance to
 Proposed Recommendation, each normative specification must have an open
 test suite of every feature defined in the specification.</p>
+
+    <p>In order for <a href="#deliverable-did-resolution">DID Resolution</a> to advance to <a href="https://www.w3.org/Consortium/Process/#RecsPR" title="Proposed Recommendation">Proposed Recommendation</a>, it is expected to define a dummy DID method meant only for demonstrating interoperability across independent implementations. While required for all complying implementations, this method will be explicitly described as unfit for use in production.
```

Comment:
I don't see why we need to define a dummy did method, just use real URLs for real controller documents, like the W3C VCWG is already proposing to do: https://w3c.github.io/vc-data-integrity/#controller-documents

See the first example in the spec:

https://w3c.github.io/vc-data-model/#example-a-simple-example-of-a-verifiable-credential

```json
{
"id": "https://university.example/issuers/565049",
"verificationMethod": [{
  "id": "https://university.example/issuers/565049#key-123",
  "type": "Multikey",
  "controller": "https://university.example/issuers/565049",
  "publicKeyMultibase": "z6MkmM42vxfqZQsv4ehtTjFFxQ4sQKS2w6WR7emozFAn5cxu"
  }]
}
```

Reply:
I guess this raises the question of if HTTPS controller documents require a "DID Method Spec"... I don't think they do, so the prohibition on defining specific DID Methods does not apply to HTTPS controller documents.

Reply:
See related issue currently blocking the VCWG: https://github.com/w3c/vc-jose-cose/issues/160

Reply:
@OR13 I'm not sure I fully understand your argument here.
> just use real URLs for real controller documents

My understanding is that DID resolvers are expecting DIDs, nor arbitrary URIs, so using "real URLs" would be out of scope. Of course, a straightforward way to define the "dummy did method" would be to naively encode a real URL in the DID.

Reply:
Thats what I am saying. The concept of "controller documents" has already expanded dids and did documents to URLs in the verifiable credentials working group.

If the DID WG does not intend to harmonize the concept of "controller document" resolution with the concept of "did document" resolution, I object to the charter entirely.

Its my understanding that is the current intention (among the authors of controller document section of the data integrity proof spec, and the authors of the did core / did resolution specs).

Reply:
@OR13 you wrote
> If the DID WG does not intend to harmonize the concept of "controller document" resolution with the concept of "did document" resolution, I object to the charter entirely.

I don't see any need for harmonization, as I see is a clear separation of concerns between what vc-data-integrity says about controller document resolution and what this charter says about DID resolution. More specifically, [vc-data-integrity, Sec 4.7](https://www.w3.org/TR/vc-data-integrity/#retrieve-verification-method) says (emphasis is mine):

> Let controllerDocument be the result of dereferencing controllerDocumentUrl, **according to the rules of the URL scheme** and using the supplied options. 

My reading is that, if the URL starts with `http:` or `https:` implementers should apply RFC9110; if it starts with `did:`, implementers should apply DID Resolution; etc. @msporny please chime in if my understanding is incorrect.

To summarize:
* vc-data-integrity defers to the specs governing each URL scheme for dereferencing
* DID Resolution aims to govern the dereferencing `did:` URLs

Reply:
@pchampin wrote:
> My reading is that, if the URL starts with http: or https: implementers should apply RFC9110; if it starts with did:, implementers should apply DID Resolution; etc. @msporny please chime in if my understanding is incorrect.

Yes, that is the correct interpretation, IMHO, @pchampin.

Reply:
> use real URLs for real controller documents

Is the notion of "controller document" compatible with "DID document"? I hope so... If not, I don't see how DID Resolution could fix any potential incompatibilities, since DID Resolution isn't about (re-)defining those data models...

Reply:
A DID Document is a specific profile of a Controller Document.

Reply:
Resolving this thread as it is made "outdated" by d8df047. @OR13 feel free to make a new comment if you feel that the "controller document" still needs dicusssion.